src/plugin.js

```diff
@@ -1,69 +1,80 @@
+      lowerCase: Joi.boolean().required(),
+      camelCase: Joi.boolean().when('lowerCase', { is: true, then: Joi.invalid(true) }),
+      allCaps: Joi.boolean().when('lowerCase', { is: true, then: Joi.invalid(true) }),
+      firstCap: Joi.boolean().when('lowerCase', { is: true, then: Joi.invalid(true) }),
+      deleteOriginal: Joi.boolean().optional()
     }),
     Joi.object().keys({
-      lower_case: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().required(),
-      all_caps: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
+      lowerCase: Joi.boolean().when('camelCase', { is: true, then: Joi.invalid(true) }),
+      camelCase: Joi.boolean().required(),
+      allCaps: Joi.boolean().when('camelCase', { is: true, then: Joi.invalid(true) }),
+      firstCap: Joi.boolean().when('camelCase', { is: true, then: Joi.invalid(true) }),
+      deleteOriginal: Joi.boolean().optional()
     }),
     Joi.object().keys({
-      lower_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().required(),
-      first_cap: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
+      lowerCase: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      camelCase: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      allCaps: Joi.boolean().required(),
+      firstCap: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      deleteOriginal: Joi.boolean().optional()
     }),
     Joi.object().keys({
-      lower_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().required(),
-      delete_original: Joi.boolean().optional()
+      lowerCase: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      camelCase: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      allCaps: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      firstCap: Joi.boolean().required(),
+      deleteOriginal: Joi.boolean().optional()
     })
   );
 
   var validate = Joi.validate(options, schemaOptions);
   if (validate.error) {
-    return next(validate.error);
+    return Promise.reject(validate.error);
   }
 
   plugin.ext('onRequest', function(request, reply) {
     if (!request.query || Object.keys(request.query).length === 0) {
-      return reply.continue();
+      return endPreHandler(reply);
```

Comment:
endPreHandler is that a signature of some sort? I cant see this anywhere in the code

Reply:
It is above this function on line 10